pkg/ottl/parser.go

```diff
@@ -69,6 +69,14 @@ func NewParser[K any](
 	settings component.TelemetrySettings,
 	options ...Option[K],
 ) Parser[K] {
+	if settings.Logger == nil {
```

Comment:
This shouldn't happen outside of tests correct? I'm curious about the choice to set a logger here. The other instances where the logger isn't set usually return an error condition.

Reply:
Mostly likely only testing, although the package is exported publicly and can be used everywhere. 

If it is more idiomatic I can update the function to return an error.

Reply:
Based on the rest of the codebase I would lean towards returning an error here.

Reply:
Switched to return an error